main.go

```diff
@@ -128,34 +130,47 @@ func generate(in configInput) ([]byte, error) {
 				ChannelCreds: []creds{
 					{Type: "google_default"},
 				},
+				ServerFeatures: []string{xdsV3ServerFeature},
 			},
 		},
 		Node: &node{
-			Id:      uuid.New().String() + "~" + in.ip,
-			Cluster: "cluster", // unused by TD
-			Locality: &locality{
-				Zone: in.zone,
-			},
+			// xDS v3 implementation in TD expects the projectNumber and networkName
+			// to be encoded in the id field, and the IP address to be encoded in the
+			// metadata field.
+			Id: fmt.Sprintf("projects/%d/networks/%s/nodes/%s", in.gcpProjectNumber, in.vpcNetworkName, uuid.New().String()),
 			Metadata: map[string]string{
 				"TRAFFICDIRECTOR_NETWORK_NAME":       in.vpcNetworkName,
 				"TRAFFICDIRECTOR_GCP_PROJECT_NUMBER": strconv.FormatInt(in.gcpProjectNumber, 10),
+				"INSTANCE_IP":                        in.ip,
+			},
+			Cluster: "cluster", // unused by TD
+			Locality: &locality{
+				Zone: in.zone,
 			},
 		},
 	}
 
 	for k, v := range in.metadataLabels {
 		c.Node.Metadata[k] = v
 	}
-	if in.includeV3Features {
-		// xDS v2 implementation in TD expects the projectNumber and networkName in
-		// the metadata field while the v3 implementation expects these in the id
-		// field.
-		c.Node.Id = fmt.Sprintf("projects/%d/networks/%s/nodes/%s", in.gcpProjectNumber, in.vpcNetworkName, uuid.New().String())
+
+	// We fallback to generating configs which work with the xDS v2 implementation
+	// in TD when this option is set.
+	if in.excludeV3Features {
 		// xDS v2 implementation in TD expects the IP address to be encoded in the
-		// id field while the v3 implementation expects this in the metadata.
-		c.Node.Metadata["INSTANCE_IP"] = in.ip
-		c.XdsServers[0].ServerFeatures = append(c.XdsServers[0].ServerFeatures, "xds_v3")
+		// id field.
+		c.Node.Id = uuid.New().String() + "~" + in.ip
+
+		// Remove the xds_v3 server feature.
```

Comment:
I don't see why the old code was rewritten. Only adding xds_v3 conditionally seems superior. I think this entire change can be simplified to just changing the default value of the flag to `true`.

Reply:
Done.